### PR TITLE
Use exponential backoff

### PR DESCRIPTION
**Pull request by xing-yang**
This PR uses exponential backoff instead of a fixed internal
when waiting for snapshot to be created and when retrying
the creation of VolumeSnapshotData.

### CODE REVIEW

**Comment by xing-yang** [comment]
@jingxu97 @tsmetana @rootfs

File: snapshot/pkg/controller/snapshotter/snapshotter.go [hunk 1]
```diff
@@ -620,22 +634,26 @@ func (vs *volumeSnapshotter) createVolumeSnapshotData(snapshotName string, snaps
...
-			// https://github.com/jpillora/backoff
 			// Re-Try it as errors writing to the API server are common
-			time.Sleep(createVolumeSnapshotDataInterval)
+			return false, err
 		} else {
-			break
+			return true, nil
 		}
-	}
+		return false, nil
```
Comment (rootfs) [inline snapshot/pkg/controller/snapshotter/snapshotter.go]: there is a govet error
```
snapshot/pkg/controller/snapshotter/snapshotter.go:655: unreachable code
exit status 1
FAILED   /home/travis/gopath/src/github.com/kubernetes-incubator/external-storage/repo-infra/verify/go-tools/verify-govet.sh	3s
```

Comment (xing-yang) [reply snapshot/pkg/controller/snapshotter/snapshotter.go]: Fixed the govet error.  Thanks.

**Comment by tsmetana** [comment]
Looks OK now. I'll try to test it.

**Comment by xing-yang** [comment]
@jingxu97 Regarding "If updateVolumeSnapshot failed once, should we break the loop?", you mean we should retry if it failed for the first time, but return failure if it failed for the 2nd time?

**Pull request merged**